crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -59,48 +59,61 @@ enum DecomposerType {
 
 struct DecomposerElement {
     decomposer: DecomposerType,
-    gate: NormalOperation,
+    gate: StandardGate,
+    params: SmallVec<[Param; 3]>,
 }
 
 #[derive(Clone, Debug)]
 struct TwoQubitUnitarySequence {
     gate_sequence: TwoQubitGateSequence,
-    decomp_gate: NormalOperation,
+    decomp_gate: StandardGate,
+    decomp_params: SmallVec<[Param; 3]>,
 }
 
 // Used in get_2q_decomposers. If the found 2q basis is a subset of GOODBYE_SET,
 // then we know TwoQubitBasisDecomposer is an ideal decomposition and there is
 // no need to bother trying the XXDecomposer.
 static GOODBYE_SET: [&str; 3] = ["cx", "cz", "ecr"];
 
+fn get_euler_basis_set(basis_list: IndexSet<&str>) -> EulerBasisSet {
```

Comment:
A small question about the type of `basis_list`: why does it need to be a `set` (and not say a `vec`) and why does it need to be `index` (the order is not important here, right?)

Reply:
I went for `IndexSet` because:
- the `set` allows to compute the intersection (used in `get_2q_decomposer_from_basis`)
- the `index` ensures that the behavior of the pass is reproducible, when we "pop" an item from the set (done in `get_2q_decomposer_from_basis` and a few more places), the item is always the same, so that the behavior is reproducible. We had a couple of issues with unit tests because the previous behavior of UnitarySynthesis was unpredictable and we decided to make it all reproducible.

If you have any suggestions I am happy to look into them.